apis/refs/v1alpha1/organizationref.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type OrganizationRef struct {
+	/* The Organization selfLink, when not managed by Config Connector. */
+	External string `json:"external,omitempty"`
+	/* The `name` field of an `Organization` resource. */
+	Name string `json:"name,omitempty"`
+	/* The `namespace` field of an `Organization` resource. */
+	Namespace string `json:"namespace,omitempty"`
+}
+
+type Organization struct {
+	OrganizationName string
+}
+
+func (s *Organization) FullyQualifiedName() string {
+	return "organizations/" + s.OrganizationName
+}
+func ResolveOrganization(ctx context.Context, reader client.Reader, src client.Object, ref *OrganizationRef) (*Organization, error) {
```

Comment:
Can we implement the `NormalizeExternal` instead of ResolveOrganization?   Resolve{{Kind}} is the old format and have  problems when integrating with its referred code